jwst/lib/suffix.py

```diff
@@ -330,11 +330,11 @@ def find_suffixes():
     # Instantiate Steps/Pipelines from their configuration files.
     # Different names and suffixes can be defined in this way.
     # Note: Based on the `collect_pipeline_cfgs` script
-    config_path = path.join(jwst_fpath, "pipeline")
-    for config_file in listdir(config_path):
-        if config_file.endswith(".cfg"):
+    config_path = jwst_fpath / "pipeline"
+    for config_file in config_path.iterdir():
+        if config_file.suffix == ".cfg":
             try:
-                step = Step.from_config_file(path.join(config_path, config_file))
+                step = Step.from_config_file(config_path / config_file)
```

Comment:
stpipe docstring claims they take "path" but cannot tell for sure from quick glance at upstream code.